examples/jsm/renderers/nodes/core/NodeBuilder.js

```diff
@@ -0,0 +1,166 @@
+const VERSION = '1';
+
+class NodeBuilder {
+
+	constructor() {
+		
+		this.slots = { vertex: [], fragment: [] };
+		this.defines = { vertex: {}, fragment: {} };
+		
+		this.nodesData = new WeakMap();
+		
+		this.shader = undefined;
+		
+	}
+	
+	setMaterial( material ) {
+		
+		this.material = material;
```

Comment:
Please define `material` in the constructor, too. That makes it easier to understand what properties the class is working with.